your_daily_dose_malware/backends/malshare.py

```diff
@@ -0,0 +1,46 @@
+import os
+
+import requests
+import click
+from dotenv import load_dotenv
+from rich.progress import Progress
+
+load_dotenv()
+
+
+def scrap_malshare():
+    with Progress() as progress:
+        api_key = os.getenv("MALSHARE_API_KEY")
+        print(api_key)
+        try:
+            response = requests.post(
+                "https://malshare.com/api.php",
+                verify=True,
+                params={"api_key": api_key, "action": "getlist"},
+            )
+            response.raise_for_status()
+            hashes = response.json()
+        except Exception as err:
+            raise err
+        try:
+            print(f"  - Malshare: {len(hashes)} Samples")
+            sha256_ids = [hashe["sha256"] for hashe in hashes]
+            task = progress.add_task(
+                "  - [green]Downloading Malware Files...", total=len(sha256_ids)
+            )
+            click.echo(f"	Your query yield {len(sha256_ids)} results")
+            for sha256_id in sha256_ids:
+                response = requests.post(
+                    "https://malshare.com/api.php",
+                    params={"api_key": api_key, "action": "getfile", "hash": sha256_id},
+                    verify=True,
+                )
+                response.raise_for_status()
+                with open(f"malware_{sha256_id[:4]}.zip", "wb") as f:
```

Comment:
Lets increase this to the last 6 chars, to prevent collisions (onboth backends). Can you also add the date of the download into the fillename? For both backends.

Reply:
@rothoma2 whenever we're writing should it be under a constant path like `resources/<backend-name>/<downloaded-file-name>`? I believe it'd make it easier to cleanup also later for user and us rather than putting it in current path.

Reply:
looks great i will implement it

Reply:
@rothoma2 what r you thinking about this ?

Reply:
but i think the goal is to collect all malwares from all thoses websites so i dont think is a good idea to create a tree like you do ... what about a simple directory like : `downloaded-malwares`

Reply:
Lets create a downloaded malware folder on the current directory, when we introduce entrypoints it i make more sense.

Maybe we should split on folders based on the filetype collected. I think that will be useful for the users split based on  lets say, exe, vba, xls etc. The only issue is when we collect the malware sample is zipped so we dont know. Maybe in the future we can think about unzipping to collect metadata.

Reply:
Sound good but when  we will implement all backends we could think about a way to to do this...